src/Clock.js

```diff
@@ -0,0 +1,103 @@
+          <div
+            className="min_hand"
+            style={{
+              transform: `rotateZ(${this.state.time.getMinutes() * 6}deg)`
+            }}
+          />
+          <div
+            className="sec_hand"
+            style={{
+              transform: `rotateZ(${this.state.time.getSeconds() * 6}deg)`
+            }}
+          />
+
+          {
+            this.state.hours ?
+              <>
+                <span className="twelve" onClick={()=>this.setState({hours: false,hour: 12})}>12</span>
+                <span className="one" onClick={()=>this.setState({hours: false,hour: 1})}>1</span>
+                <span className="two" onClick={()=>this.setState({hours: false,hour: 2})}>2</span>
+                <span className="three" onClick={()=>this.setState({hours: false,hour: 3})}>3</span>
+                <span className="four" onClick={()=>this.setState({hours: false,hour: 4})}>4</span>
+                <span className="five" onClick={()=>this.setState({hours: false,hour: 5})}>5</span>
+                <span className="six" onClick={()=>this.setState({hours: false,hour: 6})}>6</span>
+                <span className="seven" onClick={()=>this.setState({hours: false,hour: 7})}>7</span>
+                <span className="eight" onClick={()=>this.setState({hours: false,hour: 8})}>8</span>
+                <span className="nine" onClick={()=>this.setState({hours: false,hour: 9})}>9</span>
+                <span className="ten" onClick={()=>this.setState({hours: false,hour: 10})}>10</span>
+                <span className="eleven" onClick={()=>this.setState({hours: false,hour: 11})}>11</span>
+
+              </>
+            :
+              <>
+                <span className="twelve" onClick={()=>this.setState({hours: true,minute: '00'})}>0</span>
+                <span className="one" onClick={()=>this.setState({hours: true,minute: '05'})}>5</span>
+                <span className="two" onClick={()=>this.setState({hours: true,minute: '10'})}>10</span>
+                <span className="three" onClick={()=>this.setState({hours: true,minute: '15'})}>15</span>
+                <span className="four" onClick={()=>this.setState({hours: true,minute: '20'})}>20</span>
+                <span className="five" onClick={()=>this.setState({hours: true,minute: '25'})}>25</span>
+                <span className="six" onClick={()=>this.setState({hours: true,minute: '30'})}>30</span>
+                <span className="seven" onClick={()=>this.setState({hours: true,minute: '35'})}>35</span>
+                <span className="eight" onClick={()=>this.setState({hours: true,minute: '40'})}>40</span>
+                <span className="nine" onClick={()=>this.setState({hours: true,minute: '45'})}>45</span>
+                <span className="ten" onClick={()=>this.setState({hours: true,minute: '50'})}>50</span>
+                <span className="eleven" onClick={()=>this.setState({hours: true,minute: '55'})}>55</span>
+              </>
```

Comment:
Can't it be done with a `for` loop? Also, for semantics, if it's clickable, making it a `button` would be preferable.